.github/workflows/ghcr-image-upload.yml

```diff
@@ -0,0 +1,112 @@
+on:
+  pull_request:
+  push:
+    branches: [master]
+jobs:
+  build:
+    outputs:
+      image_tag: ${{ steps.get_image_tag.outputs.image_tag }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - run: docker login --username '${{ github.actor }}' --password-stdin ghcr.io <<< "$GHCR_TOKEN"
+        env:
+          GHCR_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - run: docker buildx create --driver docker-container --use
+
+      - name: build
+        run: |
+          set -euxo pipefail
+
+          args=()
+
+          if [ ${{ github.event_name }} = 'push' ]; then
+            args+=(
+              --tag ghcr.io/getsentry/launchpad:latest
+              --tag ghcr.io/getsentry/launchpad:nightly
+              --tag ghcr.io/getsentry/launchpad:amd64-latest
+              --cache-to type=registry,ref=ghcr.io/getsentry/launchpad:buildcache,mode=max
+              --push
+            )
+          fi
+
+          if [ ${{ github.event.pull_request.head.repo.full_name }} = ${{ github.repository }} ]; then
+            # TODO: we should use github artifacts to send images between jobs, like in ci.yml
+            # otherwise third-party PRs will break
+            args+=(
+              --push
+            )
+          fi
+
+          docker buildx build \
+            --pull \
+            --platform linux/amd64 \
+            --cache-from type=registry,ref=ghcr.io/getsentry/launchpad:buildcache \
+            --cache-from type=registry,ref=ghcr.io/getsentry/launchpad:latest \
+            --tag ghcr.io/getsentry/launchpad:${{ github.sha }} \
+            --target application \
+            "${args[@]}" \
+            .
+
+      - id: get_image_tag
+        run: echo 'image_tag=ghcr.io/getsentry/launchpad:${{ github.sha }}' >> "$GITHUB_OUTPUT"
+
+  publish-to-dockerhub:
```

Comment:
Let's also leave this part out for now as well, dockerhub is strictly for self-hosted use so once we get launchpad working alongside sentry we can revisit here